### PR TITLE
Feature/encapsulate cp services

### PR DESCRIPTION
**Pull request by fragkp**
Hey!

**Problem summary:**
Currently, the control panel uses the default auth guard and the 'web' middleware group. These settings are good by default, but when you change it, the control panel will be broken.

**What I've done:**
```php
if (! $this->app->runningInConsole() && Statamic::isCpRoute()) {
    Auth::shouldUse(config('statamic.cp.guard'));
}
``` 
This is the most important change. It will make sure to use always when you enter the control panel it uses the configured guard.

Also, I created a new middleware group 'statamic-cp', which has all needed middlewares by itself.

Issue references: https://github.com/statamic/cms/issues/1397 https://github.com/statamic/cms/issues/1418

### CODE REVIEW

**Comment by jasonvarga** [comment]
This is great, thanks! We'll review as soon as we can.

File: routes/web.php [hunk 1]
```diff
@@ -38,11 +40,12 @@
     }
 });
 
-Statamic::additionalWebRoutes();
+Route::middleware(Statamic::webMiddleware())->group(function () {
+    Statamic::additionalWebRoutes();
```
Comment (jasonvarga) [inline routes/web.php:44]: Did you have a particular reason to wrap this in the middleware stack too?

(Excuse the delay)

Comment (fragkp) [reply routes/web.php:44]: No particular reason, but i thought that additional routes should also have these middlewares (Localize & Cache).

**Comment by jasonvarga** [comment]
Hi Kevin

After reviewing this, it brought up some other questions that we'd want to think about before implementing this. Specifically the auth stuff. The middleware parts of this are good (although we had some tweaks to the front-end side in mind)

For instance, the CP isn't the only place we might want to use Statamic auth. How would we handle these situations? eg, the [user:login_form](https://statamic.dev/tags/user-login_form) uses auth outside the CP.

I'm gonna merge this so you get contribution credit and don't want your effort to be wasted, but just be aware that I'm gonna remove the auth stuff.

**Comment by jasonvarga** [comment]
Oh also, why was the `Auth::shouldUse` wrapped in a `! $this->app->runningInConsole()` condition?

**Comment by fragkp** [comment]
user:login_form shouldn't use the CP auth guard. That's why this PR exists.
The CP auth guard should be configurable separate from the "page" auth guards.

But, you're right. user:login_form should also be configurable. I'm not an expert for statamic (first time user) - so, maybe there are more places.

> Oh also, why was the Auth::shouldUse wrapped in a ! $this->app->runningInConsole() condition?

The request and with that, auth is not available inside commands. I'm not 100% sure if this is necessary (too long ago :D).
If I can make some free time, I will dig into that.

**Pull request closed without merging**